### PR TITLE
add functionality to hide all search indicator arrows via config

### PR DESCRIPTION
**Pull request by kalami**
config/initializer/ransack.rb

Ransack.configure do |c|
  c.remove_search_order_indicators = true
end

This is basically like passing 'hide_indicator: true' to all of your search indicators..without passing it to all of your search indicators.

### CODE REVIEW

File: lib/ransack/helpers/form_helper.rb [hunk 1]
```diff
@@ -198,7 +198,7 @@ def default_sort_order(attr_name)
           end
 
           def order_indicator
-            if @hide_indicator || no_sort_direction_specified?
+            if @hide_indicator || no_sort_direction_specified? || Ransack.options[:remove_search_order_indicators]
```
Comment (jonatack) [inline lib/ransack/helpers/form_helper.rb]: Perhaps better to do this in #initialize where @hide_indicator is decided?

File: lib/ransack/configuration.rb [hunk 2]
```diff
@@ -8,7 +8,8 @@ module Configuration
     self.predicates = {}
     self.options = {
       :search_key => :q,
-      :ignore_unknown_conditions => true
+      :ignore_unknown_conditions => true,
+      :remove_search_order_indicators => false
     }
```
Comment (jonatack) [inline lib/ransack/configuration.rb]: Perhaps call it `:remove_sort_order_indicators`, WDYT?

File: README.md [hunk 3]
```diff
@@ -213,6 +213,13 @@ The sort link may be displayed without the order indicator arrow by passing
 <%= sort_link(@q, :name, hide_indicator: true) %>
 ```
 
+Alternatively, all sort links may be displayed without the order indicator arrow by adding this to an initializer like config/initializers/ransack.rb:
+
```
Comment (jonatack) [inline README.md]: Please wrap lines at 80 characters.

File: README.md [hunk 4]
```diff
@@ -213,6 +213,13 @@ The sort link may be displayed without the order indicator arrow by passing
...
+    Ransack.configure do |c|
```
Comment (jonatack) [inline README.md]: For code formatting, use a block like:

```ruby
  code
```

File: README.md [hunk 5]
```diff
@@ -213,6 +213,13 @@ The sort link may be displayed without the order indicator arrow by passing
...
+      c.remove_search_order_indicators = true
+    end
+
```
Comment (jonatack) [inline README.md]: Please use markdown ruby code format block here.

```ruby
  code
```

**Comment by jonatack** [comment]
Thanks! Looks pretty good. I made a few comments in the code.

**Comment by kalami** [comment]
You bet!  I'm glad you like it!

File: README.md [hunk 6]
```diff
@@ -213,6 +213,15 @@ The sort link may be displayed without the order indicator arrow by passing
 <%= sort_link(@q, :name, hide_indicator: true) %>
 ```
 
+Alternatively, all sort links may be displayed without the order indicator arrow 
+by adding this to an initializer like `config/initializers/ransack.rb`:
+
+```ruby
+Ransack.configure do |c|
+  c.remove_search_order_indicators = true
```
Comment (jonatack) [inline README.md]: Perhaps should be updated to `c.hide_sort_order_indicators = true`?

File: lib/ransack/helpers/form_helper.rb [hunk 7]
```diff
@@ -197,8 +197,12 @@ def default_sort_order(attr_name)
             end
           end
 
+          def indicator?
+            @options.delete :hide_indicator unless Ransack.options[:hide_sort_order_indicators]
+          end
+
           def order_indicator
-            if @hide_indicator || no_sort_direction_specified?
+            if @hide_indicator || no_sort_direction_specified?
```
Comment (jonatack) [inline lib/ransack/helpers/form_helper.rb]: Can you remove the extra whitespace at the end of line 205?

We could shorten this to:

```ruby
def order_indicator
  return if @hide_indicator || no_sort_direction_specified?
  direction_arrow
end
```

File: lib/ransack/helpers/form_helper.rb [hunk 8]
```diff
@@ -94,7 +94,7 @@ def initialize(search, attribute, args, params)
           @current_dir    = existing_sort_direction
           @label_text     = extract_label_and_mutate_args!(args)
           @options        = extract_options_and_mutate_args!(args)
-          @hide_indicator = @options.delete :hide_indicator
+          @hide_indicator = indicator?
```
Comment (jonatack) [inline lib/ransack/helpers/form_helper.rb]: The `indicator?` method is a bit misnamed, it is really checking if Ransack should `hide_indicator?` and the logic in that method appears to be backward.

It might be better to inline this method in the constructor with this, WDYT?

```ruby
@hide_indicator = @options.delete(:hide_indicator) || Ransack.options[:hide_sort_order_indicators]
```

File: README.md (hunk 6 above)
Comment (jonatack) [inline README.md]: Thanks for the better formatting!

File: spec/ransack/helpers/form_helper_spec.rb [hunk 9]
```diff
@@ -399,6 +399,34 @@ module Helpers
...
+          Ransack.configure { |c| c.hide_sort_order_indicators = true }
+        end
+        subject { @controller.view_context
+          .sort_link(
+            [:main_app, Person.search(sorts: ['name desc'])],
+            :name,
+            controller: 'people'
+          )
+        }
+        it { should match /Full Name/ }
```
Comment (jonatack) [inline spec/ransack/helpers/form_helper_spec.rb]: 1. Does this test ever fail? Perhaps it should_not match `&nbsp;&#9660;`?

2. Could you please use RSpec `expect` syntax?

Comment (kalami) [inline spec/ransack/helpers/form_helper_spec.rb]: I think that if it is matching /Full Name/ it is automatically not matching either arrow.  I literally just copied/pasted that spec from the 'hide_indicator => true' spec because I figured the behavior is the same, but the implementation is different.

Comment (jonatack) [inline spec/ransack/helpers/form_helper_spec.rb]: I just tried that test with the inverse condition `hide_sort_order_indicators = false` and it still passes instead of failing.

File: lib/ransack/configuration.rb [hunk 10]
```diff
@@ -75,5 +76,9 @@ def arel_predicate_with_suffix(arel_predicate, suffix)
       end
     end
 
+    def hide_sort_order_indicators=(boolean)
+      self.options[:remove_search_order_indicators] = boolean
```
Comment (jonatack) [inline lib/ransack/configuration.rb]: This was not updated to the new name yet.

File: spec/ransack/helpers/form_helper_spec.rb (hunk 9 above)
Comment (jonatack) [inline spec/ransack/helpers/form_helper_spec.rb]: To make the test work properly, I'd suggest:
```ruby
it { should_not match /&#9660;|&#9650;/ }
```
Try it... it should pass when `hide_sort_order_indicators = true` and fail when `hide_sort_order_indicators = false`.

Comment (jonatack) [inline spec/ransack/helpers/form_helper_spec.rb]: OK for leaving the tests with RSpec `should` syntax since the rest of that file is that way. It might be better to move these 2 tests at the end of the `sort_link` tests where the other hide indicator specs are, and describe them like the other tests too, i.e. 'sort_link with...' instead of `search_form_for with...` since they relate to the sort_link method.

Comment (jonatack) [inline spec/ransack/helpers/form_helper_spec.rb]: If all this is confusing, don't worry about it and let me know, and I'll just commit it and clean it up.

Comment (kalami) [inline spec/ransack/helpers/form_helper_spec.rb]: Good catch, the spec for hide_indicator has the same problem.

File: lib/ransack/helpers/form_helper.rb [hunk 11]
```diff
@@ -94,7 +94,7 @@ def initialize(search, attribute, args, params)
           @current_dir    = existing_sort_direction
           @label_text     = extract_label_and_mutate_args!(args)
           @options        = extract_options_and_mutate_args!(args)
-          @hide_indicator = @options.delete :hide_indicator
+          @hide_indicator = @options.delete :hide_indicator || Ransack.options[:hide_sort_order_indicators]
```
Comment (jonatack) [inline lib/ransack/helpers/form_helper.rb]: `:hide_indicator` needs to be wrapped in parentheses here, otherwise it is evaluated by Ruby as `@options.delete (:hide_indicator || Ransack.options[:hide_sort_order_indicators])` which isn't what we want. It needs to be:

```ruby
          @hide_indicator = @options.delete(:hide_indicator) ||
            Ransack.options[:hide_sort_order_indicators]
```

File: spec/ransack/helpers/form_helper_spec.rb (hunk 9 above)
Comment (jonatack) [inline spec/ransack/helpers/form_helper_spec.rb]: You're right, good catch, that test wasn't working :+1:

**Comment by kalami** [comment]
I think I got everything.  Thanks for your patience!  This is my first contribution to a gem and I really appreciate the feedback and insights!

**Comment by jonatack** [comment]
All good! 

Here is what remains: To keep the git history clean and clear, we need to merge the commits into one, but only the ones relevant to the new feature, and update the change log. The change that fixes the other spec, and the change to the `order_indicator` method in form_helper.rb, which have nothing to do with the new feature, should be in separate commits. For the moment we want one clean commit without the two other changes in it, and with an updated change log. 

We have two options. You can do it, but it might be a little complicated if you haven't squashed/cherry-picked/rebased commits before (I can help you if needed). Or I can do it. The result will be the same. Let me know!

**Comment by jonatack** [comment]
P.S. It's great that working together, the bug in the other spec was found and fixed. Two pairs of eyes are better than one.

**Comment by kalami** [comment]
If you don't mind, I'd rather leave that stuff up to you.  Thanks again!

**Comment by jonatack** [comment]
I don't mind. Thanks for the contributions.

**Comment by jonatack** [comment]
P.S. It looks like you have different user names in your github account and your `.gitconfig` file. The git history has commits from both Josh Hunter <josh-hunter@am.com> and from kalami <kalami@users.noreply.github.com>.

**Comment by jonatack** [comment]
P.S. It looks like you have different user names/emails between your github account and your local `.gitconfig` file. The git history has commits from both `Josh Hunter <josh-hunter@am.com>` and from `kalami <kalami@users.noreply.github.com>`. Might be best to set them up the same.

**Comment by jonatack** [comment]
Once your `.gitconfig` is aligned, would you like to make a quick PR to fix that other test you spotted?
